### PR TITLE
Add deploy_new_adi_lambda script

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description

This PR adds a new utility script `deploy_new_adi_lambda.sh`, which should be used to redeploy the ADI lambda-related resources after a new lambda zip file has been uploaded to assessment_data_import S3 bucket.

This PR also contains a commit (https://github.com/cisagov/cyhy_amis/commit/a404dfe25c0c9b652be705f952a838bd45c72de8) that makes a small syntax change in several files to avoid triggering the following terraform 0.12.x warning:
```
Warning: Quoted type constraints are deprecated
   3:   type    = "list"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "list" and write
list(string) instead to explicitly indicate that the list elements are
strings.
```

## 💭 Motivation and Context

I updated the ADI lambda code in https://github.com/cisagov/assessment-data-import-lambda/pull/3 and @jsf9k made the excellent suggestion that a helper script (similar to [deploy_new_lambdas.sh](https://github.com/cisagov/cyhy_amis/blob/develop/terraform/deploy_new_lambdas.sh)) would be useful, so here it is.

## 🧪 Testing

The new script was tested and no issues were observed.

## 📷 Screenshots (if appropriate)
N/A

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes -> Note: No new tests are needed for this PR
- [x] All new and existing tests passed.

### CODE REVIEW

File: terraform/deploy_new_adi_lambda.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# deploy_new_adi_lambda.sh region workspace_name
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ $# -eq 2 ]
+then
+    region=$1
```
Comment (jsf9k) [inline terraform/deploy_new_adi_lambda.sh]: I don't think `region` is ever used.  It can be removed.

**Review by jsf9k — CHANGES_REQUESTED** [review]
See comment regarding an used variable in your bash script.

image

File: terraform/deploy_new_adi_lambda.sh (hunk 1 above)
Comment (dav3r) [reply terraform/deploy_new_adi_lambda.sh]: Ha, good catch, thanks! See https://github.com/cisagov/cyhy_amis/pull/246/commits/5a213187f09a1eff98626c295d135fb4ece6b201 for the change.  I was overzealous in my copy/pasting of a similar deploy script.

Comment (jsf9k) [reply terraform/deploy_new_adi_lambda.sh]: image

**Review by jsf9k — APPROVED** [review]

**Review by mcdonnnj — APPROVED** [review]
I'm digging the addition 👍
image

**Pull request merged**